rust/main/agents/validator/src/validator.rs

```diff
@@ -119,6 +120,10 @@ impl BaseAgent for Validator {
             )
             .await?;
 
+        let announced = metrics
+            .announced()
+            .with_label_values(&[&settings.origin_chain.name().to_string()]);
```

Comment:
Since we set `announced` metrics once, this code can run when validator is announced, namely line `self.announced.set(1);` will include the code above, but probably, better just to extract getting the metric and setting it into a separate method.

Reply:
I've combined the two lines into a single helper inside of core metrics. I *think* think is what you're asking for, but if I misunderstood please correct me.